### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by CalyCherkaoui**
## Milestone 3: Tic-tac-toe game with Ruby
In this milestone, we coded the game logic by creating the classes and setting up the relationships between their instances. The classes created are:
## Board

### CODE REVIEW

**Comment by davidmrtz-dev** [comment]
## Hi Team. 👨‍🚀
### Very good job so far.
However, there are some issues that need to be fixed, please check the comments below.  👨‍💻  ⬇️ 
****
**Please make the changes required and submit for re-review once you are ready!** 
****
### Status: Changes required. ♻️  
****


**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# inputs : player name(string) ** outputs:
+class Player
+  POSSIBILITIES = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]].freeze
+  attr_accessor :name, :picks_array, :totem
```
Comment (davidmrtz-dev) [inline lib/player.rb]: - [x] use attr_accessor only where's needed. I can see that the player class has is using attr_accessor for the name, but it's only set up on the constructor. Instead, use attr_reader. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: bin/main [hunk 2]
```diff
@@ -1,78 +1,65 @@
...
+init_player.call(player1, 1)
+init_player.call(player2, 2)
+
+# playing
+
+board = Board.new
+puts game.display_grid(board.grid)
+
+validate_pick = proc do |pick|
+  until board.pick_valid?(pick) && board.not_picked?(pick)
```
Comment (davidmrtz-dev) [inline bin/main:35]: There's an issue in the validation pick method, as you can see in the image below when I choose a position that's already filled I get the message "Number already picked, Try again" at this point it's correct, but when I select a free position, the position selected by the previous player gets filled with the actual player symbol.
Tip: You can print the board every time a selection or reselection of a position is executed.
Screen Shot 2020-06-18 at 9 38 39 PM

Comment (CalyCherkaoui) [reply bin/main:35]: Thank for the review! issue resolved

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Great work, congratulation :+1: Milestone Approved.
You did all changes required previously. 
I would suggest one more change to make your project more professional:
* It's better to add validation for cases when a player enters an empty name. In your game, I was able to just press `Enter` and play the game without typing any name. 

Then go ahead and merge to development.
_Goodluck in your next milestone._ *Happy coding*

**Pull request merged**